CONTRIBUTING.md

```diff
@@ -0,0 +1,27 @@
+## Contributing
+
+Thank your for contributing to this project! We welcome collaborators and expect users to follow our [code of conduct](CODE_OF_CONDUCT.md) when submitting code or comments.
+
+1. Fork the repo ( https://github.com/poanetwork/hbbft/fork ).
+2. Create your feature branch (`git checkout -b my-new-feature`).
+3. Write tests that cover your work.
+4. Run Rustfmt, Clippy, and all tests to ensure CI rules are satisfied. Correct versions and feature flags can be found in the [`.travis.yml`](https://github.com/poanetwork/hbbft/blob/master/.travis.yml) file.
+5. Commit your changes (`git commit -am 'Add some feature'`).
+6. Push to your branch (`git push origin my-new-feature`).
+7. Create a new Pull Request.
+
+### General
+
+* Commits should be one logical change that still allows all tests to pass.  We prefer smaller commits if there could be two levels of logic grouping.  The goal is to provide future contributors (including your future self) the reasoning behind your changes and allow them to cherry-pick, patch or port those changes in isolation to other branches or forks.
+* If during your PR you reveal a pre-existing bug:
+  1. Try to isolate the bug and fix it on an independent branch and PR it first.
+  2. Try to fix the bug in a separate commit from other changes:
+     1. Commit the code in the broken state that revealed the bug originally
+     2. Commit the fix for the bug.
+     3. Continue original PR work.
+
+### Pull Requests
```

Comment:
There have been non-PR contributions as issues or replies to issues. This may be especially valuable in the area of cryptography which, although stable at the moment, may undergo changes to push the boundaries of efficiency. Github also thinks issues may be contributions:
https://help.github.com/articles/why-are-my-contributions-not-showing-up-on-my-profile/

Reply:
Also above: If you find a bug and _don't_ manage to fix it, reporting it in a PR is also a valuable contribution.